traveler-swift-core/TravelerKit/Traveler.swift

```diff
@@ -235,6 +240,95 @@ public class Traveler {
         OperationQueue.main.addOperation(blockOperation)
     }
 
+    @discardableResult
+    func wishlistRemove(items: [Product], result: WishlistResult?, completion: @escaping(WishlistResult?, Error?) -> Void) -> WishlistResult? {
+        guard let travelerProfileId = session.identity else {
+            completion(nil, WishlistToggleError.unidentifiedTraveler)
+            return nil
+        }
+
+        var immediateResult = result
+        if let result = result, let items = items as? [CatalogItem] {
+            let allItemsExistInResult = items.map({ $0.id }).reduce(true) { (exists, id) -> Bool in
+                exists && result.items.values.contains(where: { (item) -> Bool in
+                    item.id == id
+                })
+            }
+
+            if !allItemsExistInResult {
+                completion(nil, WishlistResultError.itemsDontExistInResult)
```

Comment:
Wrong error type used here

theres a WishlistToggleError type that is public.

Maybe provide better naming there as well